Modernize PlatformCAAnimationWin.cpp similar to PlatformCAAnimationCocoa.mm

ddkilzer opened the pull request:
#### ec3619c5a844cf9526ae1a94a8bf64670c002194
<pre>
Modernize PlatformCAAnimationWin.cpp similar to PlatformCAAnimationCocoa.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=245088">https://bugs.webkit.org/show_bug.cgi?id=245088</a>
&lt;rdar://99830432&gt;

Reviewed by Brent Fulgham.

* Source/WebCore/platform/graphics/ca/win/PlatformCAAnimationWin.cpp:
(toCACFTimingFunction):
- Update to match toCAMediaTimingFunction() by using a
  references instead raw pointers, using checked casts, and
  adding an assert before the final return statement.
(PlatformCAAnimationWin::setTimingFunction):
- Remove unneeded UNUSED_PARAM().
- Add ASSERT(value) to check for nullptr.  This matches
  PlatformCAAnimationCocoa::setTimingFunction().
- Update toCACFTimingFunction() to pass a const reference.
(PlatformCAAnimationWin::setTimingFunctions):
- Remove unneeded UNUSED_PARAM().
- Switch to using a modern for loop.
- Update toCACFTimingFunction() to pass a const reference.

Canonical link: <a href="https://commits.webkit.org/254430@main">https://commits.webkit.org/254430@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/efbcca199a7ceddc8d0203383bc28199a1eb887b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88961 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33525 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98241 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/154625 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92955 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32030 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27625 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81322 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92776 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94590 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25433 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75935 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25372 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80301 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80325 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68339 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29821 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14366 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29552 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15349 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3105 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32991 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38290 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31681 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34449 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/platform/graphics/ca/win/PlatformCAAnimationWin.cpp:
```diff
@@ -539,13 +539,12 @@ void PlatformCAAnimationWin::copyKeyTimesFrom(const PlatformCAAnimation& value)
 
 void PlatformCAAnimationWin::setTimingFunctions(const Vector<Ref<const TimingFunction>>& value, bool reverse)
```
I wish we had named `value` to be `timingFunctions`.

ddkilzer replied:
I almost made that change, but `PlatformCAAnimationCocoa.mm` also used `value`, so I left it alone for consistency.

ddkilzer replied:
Posted PR for this here:  https://github.com/WebKit/WebKit/pull/4315

Landed here:
Committed [254460@main](https://commits.webkit.org/254460@main) (c020d23d66d1): <https://commits.webkit.org/254460@main>

brentfulgham commented on Source/WebCore/platform/graphics/ca/win/PlatformCAAnimationWin.cpp:
```diff
@@ -539,13 +539,12 @@ void PlatformCAAnimationWin::copyKeyTimesFrom(const PlatformCAAnimation& value)
-    for (size_t i = 0; i < value.size(); ++i)
-        CFArrayAppendValue(array.get(), toCACFTimingFunction(value[i].ptr(), reverse).get());
+    for (auto& timingFunction : value)
+        CFArrayAppendValue(array.get(), toCACFTimingFunction(timingFunction.get(), reverse).get());
```
I feel like there is an existing utility function to convert `Vector's` of `CFThings` to a `CFArray` of `CFThings`, but I can't find it, so this looks fine to me.

ddkilzer replied:
You're probably thinking of `createNSArray` from `<wtf/VectorCocoa.h>`, but there is no `createCFArray`, and since this is Windows C++ code, we can't use Objective-C.

Honestly, I find the syntax for `createNSArray` pretty hard to read (but maybe that's just because I'm not use to reading it frequently).  But `toCACFTimingFunction()` is called in one other place where there is no `Vector` of objects in use, so the conversion code would have to be duplicated or extracted anyway, so I really don't see the advantage here.

ddkilzer replied:
Implementing these convenience methods here:

Bug 245284: [WTF] Implement convenience functions for CFArray <-> Vector conversions
<https://bugs.webkit.org/show_bug.cgi?id=245284>
PR: <https://github.com/WebKit/WebKit/pull/4423>

brentfulgham reviewed (approved):
LGTM

Commit:
Modernize PlatformCAAnimationWin.cpp similar to PlatformCAAnimationCocoa.mm
https://bugs.webkit.org/show_bug.cgi?id=245088
<rdar://99830432>

Reviewed by Brent Fulgham.

* Source/WebCore/platform/graphics/ca/win/PlatformCAAnimationWin.cpp:
(toCACFTimingFunction):
- Update to match toCAMediaTimingFunction() by using a
  references instead raw pointers, using checked casts, and
  adding an assert before the final return statement.
(PlatformCAAnimationWin::setTimingFunction):
- Remove unneeded UNUSED_PARAM().
- Add ASSERT(value) to check for nullptr.  This matches
  PlatformCAAnimationCocoa::setTimingFunction().
- Update toCACFTimingFunction() to pass a const reference.
(PlatformCAAnimationWin::setTimingFunctions):
- Remove unneeded UNUSED_PARAM().
- Switch to using a modern for loop.
- Update toCACFTimingFunction() to pass a const reference.

Canonical link: https://commits.webkit.org/254430@main
